### DISCUSSION TITLE
Refctor: module singletons and globals

### DISCUSSION THREAD

**Discussion by zhuiks** [ezra-bible-app/ezra-bible-app#257] [Development]
Currently, a lot of global variables are present in the code.
Those could (should!) be refactored to be imported/required via modules. It's much easier to maintain code with very few globals.

Any JavaScript module as long as it resolves to the same filename acts like a singleton and keeps its state via closure. A good resource about that is [here](https://github.com/getify/You-Dont-Know-JS/blob/2nd-ed/scope-closures/ch8.md)

In that case appController doesn't need to hold different controller instances.

Similar there no need to encapsulate controllers and helpers in singleton classes as they are already encapsulated by modules.

**Comment by tobias-klein** [comment]
Thanks for bringing this up. I'll do my own research and then reply in more detail.

**Reply by tobias-klein** [reply]
One module (e.g. AppController) usually still needs to take the responsibility for initialization of the classes. But I guess we could consider moving that initialization into the respective modules, export the instance and then simply use require wherever we need the instance.
